cmd/codegen/generator/go/templates/module_interfaces.go

```diff
@@ -570,11 +571,17 @@ func (spec *parsedIfaceType) concreteMethodExecuteQueryCode(method *funcTypeSpec
 			s.Var().Id("results").Index().Add(underlyingReturnTypeCode).Line()
 			s.For(List(Id("_"), Id("idResult")).Op(":=").Range().Id("idResults")).BlockFunc(func(g *Group) {
 				g.Id("id").Op(":=").Id("idResult").Dot("Id").Line()
-				g.Id("results").Op("=").Append(Id("results"), Op("&").Add(underlyingImplTypeCode).Values(Dict{
-					Id("id"): Op("&").Id("id"),
-					Id("q"):  Id("querybuilder").Dot("Query").Call().Dot("Select").Call(Lit(loadFromIDQueryName)).Dot("Arg").Call(Lit("id"), Id("id")),
-					Id("c"):  Id("r").Dot("c"),
-				}))
+				d := Dict{
+					Id("Query"):  Id("querybuilder").Dot("Query").Call().Dot("Select").Call(Lit(loadFromIDQueryName)).Dot("Arg").Call(Lit("id"), Id("id")),
+					Id("Client"): Id("r").Dot("Client"),
+				}
+				if strings.Contains(loadFromIDQueryName, "IfaceFromID") {
+					// XXX: do a better hack here
+					// ...but also, this *clearly* isn't right - it feels like
+					// schemas aren't being served back
+					d[Id("id")] = Op("&").Id("id")
+				}
```

Comment:
So if I remove this (essentially the `id` would be fetched later, right), I get this bizarre error in `TestModuleIfaceBasic`:

```
    multi.go:85: 27: [41.4s]         	            	
    multi.go:85: 27: [41.4s]         	            	error: parse selections: parse field "loadOtherIfaceFromID": Query has no such field: "loadOtherIfaceFromID"
    multi.go:85: 27: [41.4s]         	Test:       	TestIface/dynamicOtherIfaceByIfaceList
```

So I thought that https://github.com/dagger/dagger/pull/6219 would serve it's own bindings to itself? It seems like it's not, and something is wrong here.